### PR TITLE
Jetpack Onboarding: Indicate if contact form is present

### LINKED ISSUE #21684: Jetpack Onboarding: Use current data in all steps
Since we can load the onboarding data at the time we show each of the steps, it makes sense to populate the data in each of the steps.

That's pretty straightforward for the Site Title and Business Address steps, where we just populate the form fields.

However, for the tile steps (Site Type, Homepage Format, Contact Form), do we want to display the current selection in any way?

Also, for the WooCommerce step, do we want to indicate in any way if WooCommerce is already installed?

cc @MichaelArestad

### PR DESCRIPTION
**Pull request by tyxla**
This PR is part of #21684 - it updates the contact form step to display a different message if the form has already been created.

**Preview (when contact form hasn't been created yet):**

**Preview (when contact form is created already):**

Fixes #21684.

To test:
* Start the onboarding flow for a fresh site.
* On the contact form step, verify it displays the "Add a contact form" button and the "Not sure? You can skip this step and add a contact form later." message below.
* Click the tile to add a contact form.
* Go back to the contact form step and verify it looks the new way.

cc @joanrho @MichaelArestad for opinion, as this particular case was not discussed in #21684.

### CODE REVIEW

File: client/jetpack-onboarding/steps/contact-form.jsx [hunk 1]
```diff
@@ -29,9 +30,20 @@ class JetpackOnboardingContactFormStep extends React.PureComponent {
...
+		const tileProps = hasContactForm
+			? {
+					description: translate( 'Your contact form has been created.' ),
+				}
+			: {
+					buttonLabel: translate( 'Add a contact form' ),
+					description: translate(
+						'Not sure? You can skip this step and add a contact form later.'
+					),
+				};
```
Comment (ockham) [inline client/jetpack-onboarding/steps/contact-form.jsx]: Maybe consider doing it separately for each prop (`buttonLabel` and `description`) so we don't have to pass `{ ...tileProps }` which can get unwieldy quickly (as it's quite opaque as to what props it contains).

**Comment by ockham** [comment]
> Design-wise, I imagine it might make sense to skip this step altogether: This is a setup wizard meant to facilitate onboarding; a step that the user has to click through doesn't really add much value...

Just noticed that there's a counter-argument to that:
* Fresh site
* User chooses to add contact form
* Clicks back
* We still want that step, or they'll be confused!

**Comment by joanrho** [comment]
This works as expected, but I'm a bit confused.

Wouldn't it make sense to show the "Your contact form has been created" message ONLY IF the contact form was filled out? It doesn't make sense to show it if the "Create form" button or tile was clicked and no form was filled out.

Am I misunderstanding something?

**Comment by joanrho** [comment]
Suggestion for how we can clarify the difference between the Contact Form PAGE and the Business Address Map WIDGET. The business address step (step 5) doesn't automatically activate the map for the "Contact info & map (Jetpack)" widget—it only displays the business name and address in that widget in the primary sidebar area. Users need to manually check the checkbox to Show Map in the widget's settings to activate the map. So here's what I propose:

**Step 4 of 7**
**Let's shape your new site.**
Would you like to create a Contact Us page with a contact form on it?
This form will allow visitors to contact you with their name, email, website, and a message.

**Step 5 of 7**
**Add a business address.**
Enter your business address to add a widget containing your address to your website.
You can add a map based on this information and change where this widget is located later on.

cc @rickybanister

File: client/jetpack-onboarding/steps/contact-form.jsx (hunk 1 above)
Comment (tyxla) [reply client/jetpack-onboarding/steps/contact-form.jsx]: Yeah, I tried that in https://github.com/Automattic/wp-calypso/pull/22247/commits/53e55481eb4f88f27c7892263ca9345f7ad07ead and you're right that it looks better.

**Comment by tyxla** [comment]
Copy updates are handled in #22282.

**Pull request merged**